pkg/controller/controller.go

```diff
@@ -298,3 +300,20 @@ func (p *csiProvisioner) volumeIdToHandle(id string) string {
 func (p *csiProvisioner) volumeHandleToId(handle string) string {
 	return handle
 }
+
+func getCredentialsFromSecret(k8s kubernetes.Interface, sn, ns string) map[string]string {
```

Comment:
This function should return error if it fails during retrieval of secrets.  If the secret is set on the StorageClass and it fails here, it should not just assign empty string, it should return an error.

Reply:
@vladimirvivien please see my response to your previous comment on the same matter.